external-api/src/interfaces/number.interface.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Defines the interfaces for the response of the number generator.
+ *
+ * @interface NumberGeneratorResponse
+ * @property {number} status - The status code of the response.
+ * @property {Array} data - An array of objects containing the generated numbers.
+ * @property {number} data.id - The ID of the generated number.
+ * @property {number} data.maskedId - The masked ID of the generated number.
+ * @property {number} data.type - The type of the generated number.
+ * @property {string} data.createdBy - The creator of the generated number.
+ * @property {string} data.createdDatetime - The creation datetime of the generated number.
+ * @property {string} data.requestingSystem - The system that requested the generation of the number.
+ */
+export interface NumberGeneratorResponse {
+  readonly status: number;
+  readonly data: Array<{
+    id: number;
+    maskedId: number;
+    type: number;
+    createdBy: string;
+    createdDatetime: string;
```

Comment:
I wouldn't usually do this, but I've just come off the back of having to track down what format a datetime was. A string datetime could be uniquely formatted/iso formatted/string epoch/ string epoch ms etc etc. 

This didn't particularly used to matter, as moment was surprisingly relaxed about what it could parse as a date. Datefns is a bit more strict (rightly so) about what kind of dates it will pass.

As a result -- It really helps to document this while it's fresh. I can see it's done in tests, but could we also use the type alias of `IsoDateTimeStamp` -- This is already used elsewhere in the project :)

```
/**
 * ISO 8601 date time string in format 'yyyy-MM-ddThh:mm:ssZ'
 */
export type IsoDateTimeStamp = string;
```

Reply:
Good shout, have added.